internal/provider/nats.go

```diff
@@ -0,0 +1,75 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/nats-io/nats.go"
+	"github.com/shopmonkeyus/eds-server/internal"
+	"github.com/shopmonkeyus/go-common/logger"
+	"github.com/shopmonkeyus/go-datamodel/datatypes"
+)
+
+type NatsProvider struct {
+	logger logger.Logger
+	nc     *nats.Conn
+	opts   *ProviderOpts
+}
+
+var _ internal.Provider = (*NatsProvider)(nil)
+
+// NewFileProvider returns a provider that will stream files to a folder provided in the url
+func NewNatsProvider(logger logger.Logger, urlstring string, opts *ProviderOpts) (internal.Provider, error) {
+	nc, err := nats.Connect(urlstring)
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to nats server: %s", urlstring)
+	}
+
+	logger.Info("nats provider will publish to: %s", urlstring)
+	return &NatsProvider{
+		logger,
+		nc,
+		opts,
+	}, nil
+}
+
+// Start the provider and return an error or nil if ok
+func (p *NatsProvider) Start() error {
+	return nil
+}
+
+// Stop the provider and return an error or nil if ok
+func (p *NatsProvider) Stop() error {
+	p.nc.Close()
+	return nil
+}
+
+// Process data received and return an error or nil if processed ok
+func (p *NatsProvider) Process(data datatypes.ChangeEventPayload) error {
+	p.logger.Info("Republish Message to: dbchange.%s.%s.%s",  data.GetTable(), data.GetOperation(), *data.GetLocationID())
+	subject := fmt.Sprintf("dbchange.%s.%s.%s", data.GetTable(), data.GetOperation(), *data.GetLocationID())
```

Comment:
you'll want to add this check:

![image](https://user-images.githubusercontent.com/6027/224163976-527dda59-9ba7-457e-889c-477d9b81bc8a.png)

Reply:
This is already done prior to the messages hitting the NatsProvider Processor so anything we receive will already only be the public models.